src/entities/amm.ts

```diff
@@ -346,6 +347,13 @@ class AMM {
       this.isETH = false;
       this.isFCM = true;
     }
+
+    if (this.fcmAddress === ADDRESS_ZERO) {
+      this.isFCM = false;
+    } else {
+      this.isFCM = true;
+    }
+    
```

Comment:
I think there is something wrong with this. The way this stands, it will override the logic above it which also sets the isFCM flag. 

Perhaps you don't need the `else` clause?

Reply:
Also, its entirely possible I just don't understand what it is we're trying to achieve.